dts/bindings/serial/ambiq,uart.yaml

```diff
@@ -0,0 +1,20 @@
+# Copyright (c) 2023 Antmicro <www.antmicro.com>
+# SPDX-License-Identifier: Apache-2.0
+
+description: Ambiq UART controller (PL011 compatible)
+
+compatible: "ambiq,uart"
+
+include: ["arm,pl011.yaml", pinctrl-device.yaml]
+
+properties:
+  pinctrl-0:
+    required: true
+
+  pinctrl-names:
+    required: true
+
+  ambiq,uart-instance:
+    type: int
+    description: |
+      Instance number of the UART peripheral
```

Comment:
this has nothing to do with hw description, please remove.

Reply:
In fact, it has to do with the description of hw, they are not just logical instances. 
Using instances from Zephyr for example. `DT_INST_FOREACH_STATUS_OKAY` we will get an instance of 0 for `uart2` if only `uart2` has `status = "okay"`. By using `ambiq,uart-instance` we are sure that we are passing the correct hardware instance to the driver.

Reply:
I think we misunderstood each other. The `ambiq,uart-instance` field is specific to each uart instance in DTS. Adding this field ensures that it is properly handled by the driver regardless of the driver instance.

Reply:
Again, DT is reserved for hardware description, not driver-specific needs. Instance numbers are a common sign of halisms in Devicetree because HALs tend to rely on those numbers.

I haven't checked in detail, but this kind of functionality is usually handled by reset/clock controllers where you have in DT a cell where to encode information (e.g. the peripheral bit in enablement register).

Reply:
Ok, I get it. We'll get rid of this instance.

Reply:
Removed `ambiq,uart-instance`.